lib/ui/compositing/scene_builder.cc

```diff
@@ -149,8 +149,14 @@ fml::RefPtr<EngineLayer> SceneBuilder::pushColorFilter(
   return EngineLayer::MakeRetained(layer);
 }
 
-fml::RefPtr<EngineLayer> SceneBuilder::pushBackdropFilter(ImageFilter* filter) {
+fml::RefPtr<EngineLayer> SceneBuilder::pushBackdropFilter(
+    ImageFilter* filter,
+    bool isDynamic,
+    fml::RefPtr<EngineLayer> oldLayer) {
   auto layer = std::make_shared<flutter::BackdropFilterLayer>(filter->filter());
+  if (!isDynamic) {
+    layer.get()->initRetained(oldLayer ? oldLayer->Layer() : layer);
```

Comment:
Having an `oldLayer` does not necessarily mean that the `layer` and the `oldLayer` have the same `filter`? Maybe it's easier to let the framework send in an identifier (the version number in your doc), and use that identifier to decide which cache to use (or whether to use cache).

Reply:
I'm not sure I understand the question. I need oldLayer to have the same unique-id as when it was cached. I've already identified above that our ability to detect a change in the filter is lacking. Currently, if they specify isDynamic=false then we trust that the filter hasn't changed, but that isn't a robust solution.

Reply:
It looks like ColorFilter has a public class that simply stores the arguments and can compare against other ColorFilters and produce a pretty string. When asked (via private method), it can produce a NativeFieldWrapper version.

ImageFilter just goes straight to being the NativeFieldWrapper itself with no stored parameters that it can use to implement operator== or print a nice string. I'm guessing we want ImageFilter to behave more like ColorFilter and then our careful == tests in the framwork will detect when the filter changes appropriately. I'll implement those changes to ImageFilter as well...